chainerx_cc/chainerx/cuda/memory_pool.h

```diff
@@ -18,8 +19,14 @@ class MemoryPoolTest;  // for unit-tests
 
 }  // namespace cuda_internal
 
+// TODO(imanishi): Determine this value at runtime.
+// TODO(imanishi): Define two constant values: the unit size of allocator and uint size of memory pool.
+// CUDA aligns to multiples of 512.
 constexpr size_t kAllocationUnitSize = 512;
```

Comment:
How about something like this?
```
// Allocation unit size. This value should be a multiple of the allocation unit size of underlying CUDA allocator.
// TODO(imanishi): The unit sizes of CUDA allocators are not documented.
// It may be dependent on various factors, such as CUDA runtime version or allocation functions.
// It has been observed that `cudaMallocManaged` has an allocation unit size of 4096 bytes in certain environment.
// We should revisit this number later, or perhaps better to determine it at runtime.
// TODO(imanishi): This number is currently used as both the underlying allocation unit size and the allocation unit size of the memory pool (the unit size of splitted chunks).
// We could separate them and fine-tune to optimize to the typical memory usage.
```